SimTracker/TrackerMaterialAnalysis/plugins/TrackingMaterialProducer.cc

```diff
@@ -281,12 +283,24 @@ void TrackingMaterialProducer::update(const G4Step* step) {
   }
 
   // update track accounting
-  if (enter_sensitive)
-    m_track.enterDetector(sensitive, position, cosThetaPre);
+  if (enter_sensitive) {
+    if (m_track_volume != nullptr) {
+      edm::LogWarning("TrackingMaterialProducer") << "Entering volume " << sensitive << "while inside volume "
+                                                  << m_track_volume << ". Something is inconsistent";
+      m_track.reset();
+    }
+    m_track_volume = sensitive;
+    m_track.enterDetector(position, cosThetaPre);
+  }
   m_track.step(MaterialAccountingStep(length, radiationLengths, energyLoss, globalPositionIn, globalPositionOut));
-  if (leave_sensitive)
-    m_track.leaveDetector(sensitive, cosThetaPost);
-
+  if (leave_sensitive) {
+    if (m_track_volume != sensitive) {
+      edm::LogWarning("TrackingMaterialProducer") << "Leaving volume " << sensitive << "while inside volume "
+                                                  << m_track_volume << ". Something is inconsistent";
```

Comment:
Should it be an error/exception?

Reply:
It wasn't before. No change in behavior is proposed...

Reply:
I see, but this was handled internally by `SimDataFormats/ValidationFormats/src/MaterialAccountingTrack.cc`. I get a couple of [empty histograms](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-961ef0/8982/trackerMaterialAnalysisPlots/). FYI: @mmusich, @mtosi, @apsallid